### ISSUE TITLE
option paths=source_relative:. causes import cycle loop

### ISSUE THREAD

**Issue by strobil** [twitchtv/twirp#196]
Command used for generating *twirp.go

> protoc -I /usr/local/include/ \
>   -I $GOPATH/src \
>   -I $GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
>   -I . \
>   -I tv_service \
>   --go_out=plugins=grpc,paths=source_relative:. \
>   --grpc-gateway_out=logtostderr=true,grpc_api_configuration=tv_service/tv_service.yaml,paths=source_relative:. \
>   --twirp_out=paths=source_relative:. \
>   tv_service/tv_service.proto

When i am trying to build project this error occurs:

> import cycle not allowed
> package gitlab.sweet.tv/go/tv_service
> 	imports gitlab.sweet.tv/proto/tv_service
> 	imports gitlab.sweet.tv/proto/tv_service

The reason is in generated file which contains this line, but it should not, because the package name is identical in both cases (tv_service)

`import tv_service8 "gitlab.sweet.tv/proto/tv_service"`

If i remove paths=source_relative:. the generated file does not contain this import line.

**Comment by spenczar** [comment]
This is hard to debug without more detail. Can you share the `.proto` file you are using, or a stripped-down version of it that minimally reproduces your error?

**Comment by strobil** [comment]
@spenczar Yes, sure.

Untar these file and run go_generate.sh

[proto2.tar.gz](https://github.com/twitchtv/twirp/files/3917489/proto2.tar.gz)

**Comment by spenczar** [comment]
Thanks, I'm able to reproduce your error. I'll isolate it into a test case and work on a fix.

**Comment by spenczar** [comment]
@strobil can you test the branch 'fix_local_proto_imports' (https://github.com/twitchtv/twirp/pull/202) to confirm that it fixes your full build? It works for me locally.

**Comment by strobil** [comment]
@spenczar, how i can install a specific version of protoc-gen-twirp?

**Comment by spenczar** [comment]
@strobil Something like this:
```
export GOPATH=$(mktemp -d)
mkdir -p $GOPATH/src/github.com/twitchtv/twirp
git clone https://github.com/twitchtv/twirp.git $GOPATH/src/github.com/twitchtv/twirp
cd $GOPATH/src/github.com/twitchtv/twirp
git checkout -b fix_local_proto_imports origin/fix_local_proto_imports
go build -o $HOME/protoc-gen-twirp $GOPATH/src/github.com/twitchtv/twirp
```
Now you'll have an updated version of `protoc-gen-twirp` in `$HOME`. Add `$HOME` to the front of your `PATH`, and then `protoc` should use it.

```

**Comment by dpolansky** [comment]
Spencer's fix for this issue was released in v5.10.2.
